xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletResponse.java

```diff
@@ -75,14 +84,42 @@ public void sendRedirect(String redirect) throws IOException
                             + "the configuration: it can be configured in xwiki.properties in url.trustedDomains.",
                         redirect);
                 } else {
-                    this.response.sendRedirect(redirect);
+                    this.response.sendRedirect(uri.toString());
+                }
+            }
+        }
+    }
+
+    private URI parseURI(String location)
```

Comment:
Feels like this should also be part of an API: the reason is that I need to also parse a String to URI in the new URLSecurityScriptService, so would be useful to use the same code in all places where we have this need.

Reply:
Using this code in `URLSecurityScriptService` is a very bad idea as there will be no guarantee that the original string representation is safe when the parsed URI is safe. I know examples where this is not the case, i.e., where the original string results in a different host in a browser than what the URI has as host.

Reply:
> Using this code in URLSecurityScriptService is a very bad idea as there will be no guarantee that the original string representation is safe when the parsed URI is safe. I know examples where this is not the case, i.e., where the original string results in a different host in a browser than what the URI has as host.

Then it means the script service is useless if we cannot check the safeness of the given URL